### ISSUE TITLE
Update near-api-js dependency

### ISSUE THREAD

**Issue by bh2smith** [near/wallet-selector#1104]
The current [near-api-js](https://www.npmjs.com/package/near-api-js) version is 4, leaving this library 2 major versions behind:

https://github.com/near/wallet-selector/blob/594f530bf729b6c3d2f72a42612345e40654a9be/package.json#L112

Furthermore, the core library specifies fixed peer dependencies at v1 || v2:

https://github.com/near/wallet-selector/blob/594f530bf729b6c3d2f72a42612345e40654a9be/packages/wallet-connect/package.json#L24-L26

Is there any specific reason these are so far behind? It is causing conflicts with npm installation of any library using v3 or v4.

**Comment by bh2smith** [comment]
Someone please deal with this:
```
warning "near-ca > @near-wallet-selector/core@8.9.8" has incorrect peer dependency "near-api-js@^1.0.0 || ^2.0.0".
```

**Comment by pivanov** [comment]
This should be fixed in #1135
@bh2smith can you confirm?
